### PR TITLE
Correct case in renaming

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
The main goal is to stay close to Anki code. Furthermore, since I'm the author of the last commits in Anki, I guess it's easier for me than for anybody else to port the code to AnkiDroid.

### Bug 1.

Create decks "A::B", "b" and "C". Now rename "b" to "A::b". You now have a
deck "A::b" while you already have "A::B". Try to rename "C" to "A::b::C". It'll be moved as
"A::B::C". It seems that "A::b" can't get children.

### Bug 2.
Let's assume that you already have decks A::B and A::b. This is not detected by any correction of the collection. Now, the second deck will be renamed and user will eventually see they have to correct it.

### Bug 3.

(This bug was present in Anki and seems to be absent from Ankidroid, for some unknown reason)
Create a filtered deck "Z"
Create another deck "Y".
Rename "Y" to "z::Y".

The deck will be renamed to Z::Y. Which means that the filtered deck has child.
With my correction, this will be rejected.

## Approach
As in Anki. Doing all names comparison ignoring cases

This PR is actually based over the code of https://github.com/ankidroid/Anki-Android/pull/5601 . I decided to create two distinct PRs so that you can merge https://github.com/ankidroid/Anki-Android/pull/5601 quickly (because without it AnkiDroid bugs today). And you can take time to review the current PR and eventually accept its when you have time.
if you directly merge the current PR, then https://github.com/ankidroid/Anki-Android/pull/5601 becomes useless.

## How Has This Been Tested?

Try to reproduce both bugs. See that you fail.

## Learning (optional, can help others)
Quite a funny story here. I was trying to create an add-on. And while I was trying to understand how to do it to ensure that all comparisons are done without comparing cases, I failed. More precisely, I didn't understand how anki did it. Finally, I realized that actually, Anki did have those bugs. Which lead me to correct them. I did actually report them in the PR, since there was no point in doing a bug report and a PR separately. (Which, in a sens, means that I'm one of the few people who actually can do bug report on Anki's github instead of using the forum :p)

While correcting the bug 1, I did also correct the bug 2. Even if I didn't discover the bug 2 untill DAE asked me why the correction was so complex. It's only to be able to answer his question that I did work out how to create a bug needing a complex correction.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by mikehardy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android_2/Anki-Android_2/pullRequest?prid=4496868&bid=15162179)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java [hunk 1]
```diff
@@ -888,12 +885,13 @@ private void _checkDeckTree() {
             // immediate parent must exist
             String immediateParent = parent(deckName);
             if (immediateParent != null) {
-                if (!names.contains(immediateParent)) {
+                if (!names.contains(normalizeName(immediateParent))) {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/app/timrae/Anki-Android/pullRequest?prid=4496867)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by mikehardy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android_2/Anki-Android_2/pullRequest?prid=4496868&bid=15162179)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by mikehardy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android_2/Anki-Android_2/pullRequest?prid=4496868&bid=15162179)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by mikehardy** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android_2/Anki-Android_2/pullRequest?prid=4496868&bid=15162179)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Anki-Android/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by Arthur-Milchior** [comment]
Can you restart 25 please ? I don't get what means the API 24 JDK 1.11; it's an error I never saw before

**Comment by mikehardy** [comment]
Same emulator hang for 25, I'll have to evict it from the matrix if it continues.

You'll notice for the JDK1.11 one on other runs it is marked as "okay to fail", that is it is not essential it passes for the whole run to pass. Despite having a few areas of the app where we're still busy forward-porting from old APIs or upstream, there are others where we try to look forward and work with the future to see how we do - JDK1.11 is one of those - works in some cases but not perfectly so we don't mark it as required in travis

**Comment by mikehardy** [comment]
@Arthur-Milchior I have merged #5601 so this should be rebased to remove that commit I believe, then I/we can review this in depth as you mentioned. Thank you!

**Comment by Arthur-Milchior** [comment]
Done

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4496867&bid=15162178)

**Comment by mikehardy** [comment]
@Arthur-Milchior another one finally going in! 🍾

**Pull request merged**